packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/component_builder.rs

```diff
@@ -0,0 +1,164 @@
+                    }
+                    // Some references
+                    component.ref_transmutes_to();
+                    // XXX: we temporarily fill each required attribute with an empty vector.
+                    // This will be removed when typed attributes are integrated.
+                    let attributes: HashMap<&'static str, _> = HashMap::from_iter(
+                        component
+                            .get_attribute_names()
+                            .iter()
+                            .map(|&name| (name, Vec::new())),
+                    );
+
+                    let extending: Result<Option<ExtendSource>, anyhow::Error> =
+                        match &elm.extending {
+                            Some(ref_resolution) => {
+                                let ref_resolution = ref_resolution.get_resolution();
+                                // If there is no remaining path, we are extending a component directly. Otherwise,
+                                // we look up the state variable on that component and extend it.
+                                if let Some(unresolved_path) = &ref_resolution.unresolved_path {
+                                    if unresolved_path.len() != 1 {
+                                        Err(anyhow!("Handle nested state variables"))
+                                    } else {
+                                        // We only know how to handle single state variable access at the moment.
+                                        if let Some(state_var) = component
+                                            .get_state_variable_index_from_name(
+                                                &unresolved_path[0].name,
+                                            )
+                                        {
+                                            Ok(Some(ExtendSource::StateVar(
+                                                ExtendStateVariableDescription {
+                                                    component_idx: ref_resolution.node_idx,
+                                                    state_variable_matching: vec![
+                                                        StateVariableShadowingMatch {
+                                                            // XXX: THIS 0 IS MADE UP AND IS A PLACEHOLDER UNTIL WE HAVE AN ACTUAL METHOD
+                                                            shadowed_state_var_idx: 0,
+                                                            shadowing_state_var_idx: state_var,
+                                                        },
+                                                    ],
+                                                },
+                                            )))
+                                        } else {
+                                            Err(anyhow!(
+                                                "State variable {} not found on component {}",
```

Comment:
The name could be "" if we stopped due to a path index. Probably want a different message in that case.

Reply:
Also, if there is a path index after a matching state variable, do we want to `Err` for now?